pittgoogle/pubsub.py

```diff
@@ -404,56 +393,77 @@ def client(self) -> google.cloud.pubsub_v1.SubscriberClient:
             )
         return self._client
 
-    def touch(self) -> None:
+    def touch(self, attribute_filter: str | None, smt_javascript_udf: str | None) -> None:
         """Test the connection to the subscription, creating it if necessary.
 
         Note that messages published to the topic before the subscription was created are
         not available to the subscription.
 
+        Args:
+            attribute_filter (str, optional):
+                To filter messages, specify an expression that operates on message attributes. The expression is an
+                immutable property of a subscription. After you create a subscription, you cannot update the
+                subscription to modify the expresssion. The syntax used to create a filter is outlined in:
+                https://docs.cloud.google.com/pubsub/docs/subscription-message-filter.
+            smt_javascript_udf (str, optional):
+                Specify a JavaScript User-Defined Function (UDF), a type of Single Message Transform (SMT) that allows
+                for the implementation of custom transformation logic within Pub/Sub. UDFs attached to a subscription
+                can enable a wide range of use cases, including: message filtering (based on the message payload and/or
+                attributes), simple data transformations, data masking and redaction, and data format conversions. An
+                overview of UDFs is outlined in: https://docs.cloud.google.com/pubsub/docs/smts/udfs-overview.
+
         Raises:
             TypeError:
                 if the subscription needs to be created but no topic was provided.
-
             CloudConnectionError:
-
                 - 'NotFound` if the subscription needs to be created but the topic does not exist in Google Cloud.
                 - 'InvalidTopic' if the subscription exists but the user explicitly provided a topic that
                    this subscription is not actually attached to.
         """
         try:
             subscrip = self.client.get_subscription(subscription=self.path)
-            LOGGER.info(f"subscription exists: {self.path}")
+            LOGGER.info(
+                f"subscription exists: {self.path}. Keyword arguments are not applicable in this context."
+            )
```

Comment:
If kwargs were passed in, this needs to raise a warning and not just a simple info message. So do something like:

```python
            if attribute_filter or smt_javascript_udf:
                LOGGER.warning(<message here>)
            else:
                LOGGER.info(f"subscription exists: {self.path}")

```